src/ray/gcs/redis_context.cc

```diff
@@ -560,10 +562,11 @@ Status ConnectRedisSentinel(RedisContext &context,
   }
 }
 
-std::vector<std::string> ResolveDNS(const std::string &address, int port) {
+std::vector<std::string> ResolveDNS(instrumented_io_context &io_service,
+                                    const std::string &address,
+                                    int port) {
   using namespace boost::asio;
-  io_context ctx;
-  ip::tcp::resolver resolver(ctx);
+  ip::tcp::resolver resolver(io_service);
```

Comment:
would this call poll on the io_service in a different thread, and is this the main_io_context for the gcs. I actually just made the other pr with the concurrency hint for main_io_context which assumes io ops would only happen on a single thread